staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -16,7 +16,7 @@ limitations under the License.
 
 package v1
 
-// JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).
+// JSONSchemaProps is a JSON-Schema following Specification Draft 4. See http://json-schema.org/ for more details.
```

Comment:
```suggestion
// JSONSchemaProps is a JSON-Schema following Specification Draft 4. See https://json-schema.org/ for more details.
```
That website says:
> The current version is 2020-12! The previous version was 2019-09.

I wonder if JSONSchemaProps still follows a draft spec or if it has been updated.

Reply:
Hi @sftim this is the specification links reference page : https://json-schema.org/specification-links.html 

2020-12 would have been Draft-8 acc to previous convention. 
Might have to investigate about what is followed . Draft 4 or latest.

Reply:
Hi @sftim , Naming convention has been changed after Draft 7 and While investigating, I found the Changes from draft 4 to Draft 6 [here](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-01#appendix-B) it shows that 
-  Changed "id" to "$id"; all core keywords prefixed with "$"


However our file has this [here](https://github.com/kubernetes/kubernetes/blob/ec39cc2eafffa51b3267e3bd64fbd2598c0db94d/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go#L21) on L21.
 -	ID          string        `json:"id,omitempty" protobuf:"bytes,1,opt,name=id"`
 
Perhaps, the JSONSchemaProps still follows Draft spec and not the latest.

Did I get it right?

Reply:
Also http://json-schema.org/draft-06/json-schema-release-notes.html , we can see the changes from Draft 4 to 6 and our [code](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go#L21) still has the field types changed here in Draft 6.